sktime/forecasting/base/_base.py

```diff
@@ -631,6 +637,48 @@ def update_predict_single(
             alpha=alpha,
         )
 
+    def predict_residuals(self, y, X=None, fh=None):
+        """Return residuals of time series forecasts.
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_".
+            self.cutoff, self._is_fitted
+
+        Writes to self:
+            Stores fh to self.fh if fh is passed and has not been passed previously.
+
+        Parameters
+        ----------
+        y : pd.Series, pd.DataFrame, or np.ndarray (1D or 2D)
+            Time series with ground truth observations, to compute residuals to.
+            Must have same type, dimension, and indices as expected return of predict.
+        X : pd.DataFrame, or 2D np.ndarray, optional (default=None)
+            Exogeneous time series to predict from
+            if self.get_tag("X-y-must-have-same-index"), X.index must contain fh.index
+        fh : int, list, np.ndarray or ForecastingHorizon
+            Forecasting horizon
+
+        Returns
+        -------
+        y_res : pd.Series, pd.DataFrame, or np.ndarray (1D or 2D)
+            Forecast residuals at fh, with same index as fh
+            y_pred has same type as y passed in fit (most recently)
+        """
+        y_pred = self.predict(fh=fh, X=X)
```

Comment:
@fkiraly I do not see the handling for `fh=None` (i.e. return all in-sample residuals). Is that something that will be added later?

Reply:
we could do this in this PR if you'd prefer that? I wouldn't mind.
Also not sure what the other core devs think what the default behaviour for `fh=None` should be.

Reply:
ok, done - see below (if no argument is passed)